compiler/rustc_codegen_ssa/src/back/archive.rs

```diff
@@ -307,3 +308,29 @@ impl<'a> ArArchiveBuilder<'a> {
 fn io_error_context(context: &str, err: io::Error) -> io::Error {
     io::Error::new(io::ErrorKind::Other, format!("{context}: {err}"))
 }
+
+pub fn read_archive_file_undefined_symbols(archive_path: &Path) -> io::Result<Vec<String>> {
+    let archive_map = unsafe { Mmap::map(File::open(&archive_path)?)? };
+    let archive = ArchiveFile::parse(&*archive_map)
+        .map_err(|err| io::Error::new(io::ErrorKind::InvalidData, err))?;
+    let mut undefined_symbols = Vec::new();
+    for entry in archive.members() {
+        let entry = entry.map_err(|err| io::Error::new(io::ErrorKind::InvalidData, err))?;
+        let data = entry
+            .data(&*archive_map)
+            .map_err(|err| io::Error::new(io::ErrorKind::InvalidData, err))?;
+        let file = object::File::parse(&*data)
+            .map_err(|err| io::Error::new(io::ErrorKind::InvalidData, err))?;
+        for symbol in file.symbols() {
+            if symbol.is_undefined() {
```

Comment:
This doesn't work for webassembly.

Reply:
Resolved, new related issues I put in TODO.